File: answers/leetcode/645. Set Mismatch/Solution.java
class Solution {
    public int[] findErrorNums(int[] nums) {
        int[] error = new int[2];
        for (int num : nums) {
            int abs = Math.abs(num);
            if (nums[abs - 1] > 0) {
                nums[abs - 1] *= -1;
            } else {
                error[0] = abs;
            }
        }
        for (int i = 0; i < nums.length; i++) {
            if (nums[i] > 0) {
                error[1] = i + 1;
            }
        }
        return error;
    }
}

File: answers/leetcode/645. Set Mismatch/Solution.py
class Solution:
    def findErrorNums(self, nums: List[int]) -> List[int]:
        repetition = -1
        loss = -1
        number_set = set()
        for number in nums:
            if number in number_set:
                repetition = number
            else:
                number_set.add(number)
        for i in range(len(nums)):
            if i + 1 not in number_set:
                loss = i + 1
        return [repetition, loss]
